std/experimental/allocator/building_blocks/free_list.d

```diff
@@ -359,14 +359,27 @@ struct FreeList(ParentAllocator,
     }
 
     /**
-    Defined only if $(D ParentAllocator) defines $(D deallocateAll). If so,
-    forwards to it and resets the freelist.
+    If the parent allocator implements `deallocateAll` then forwards it otherwise
+    each element is freed with the parent's `deallocate`.
     */
-    static if (hasMember!(ParentAllocator, "deallocateAll"))
     bool deallocateAll()
     {
+        bool result = false;
+        static if (hasMember!(ParentAllocator, "deallocateAll"))
+        {
+            result = parent.deallocateAll();
+        }
+        else static if (hasMember!(ParentAllocator, "deallocate"))
+        {
+            result = true;
+            for (auto n = root; n; n = n.next)
+            {
+                if (!parent.deallocate((cast(ubyte*)n)[0 .. max]))
+                    result = false;
+            }
+        }
```

Comment:
Added the static assert. One question I didn't asked before is why the hell there is no template used as constraint for the different allocators ?  
A bit like _isInputRange_ eg

```D
template isMinimalAllocator(T)
{
    enum isMinimalAllocator = /*static checks*/...
}
```